src/cascadia/UnitTests_TerminalCore/ConptyRoundtripTests.cpp

```diff
@@ -0,0 +1,366 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+// This test class creates an in-proc conpty host as well as a Terminal, to
+// validate that strings written to the conpty create the same resopnse on the
+// terminal end. Tests can be written that validate both the contents of the
+// host buffer as well as the terminal buffer. Everytime that
+// `renderer.PaintFrame()` is called, the tests will validate the expected
+// output, and then flush the output of the VtEngine straight to the Terminal.
+
+#include "precomp.h"
+#include <wextestclass.h>
+#include "../../inc/consoletaeftemplates.hpp"
+#include "../../types/inc/Viewport.hpp"
+#include "../../types/inc/convert.hpp"
+
+#include "../renderer/inc/DummyRenderTarget.hpp"
+#include "../../renderer/base/Renderer.hpp"
+#include "../../renderer/vt/Xterm256Engine.hpp"
+#include "../../renderer/vt/XtermEngine.hpp"
+#include "../../renderer/vt/WinTelnetEngine.hpp"
+
+class InputBuffer; // This for some reason needs to be fwd-decl'd
+#include "../host/inputBuffer.hpp"
+#include "../host/readDataCooked.hpp"
+#include "test/CommonState.hpp"
+
+#include "../cascadia/TerminalCore/Terminal.hpp"
+
+using namespace WEX::Common;
+using namespace WEX::Logging;
+using namespace WEX::TestExecution;
+using namespace Microsoft::Console::Types;
+using namespace Microsoft::Console::Interactivity;
+using namespace Microsoft::Console::VirtualTerminal;
+
+using namespace Microsoft::Console;
+using namespace Microsoft::Console::Render;
+using namespace Microsoft::Console::Types;
+
+using namespace Microsoft::Terminal::Core;
+
+class ConptyRoundtripTests
+{
+    TEST_CLASS(ConptyRoundtripTests);
+
+    TEST_CLASS_SETUP(ClassSetup)
+    {
+        m_state = std::make_unique<CommonState>();
+
+        m_state->InitEvents();
+        m_state->PrepareGlobalFont();
+        m_state->PrepareGlobalScreenBuffer();
+        m_state->PrepareGlobalInputBuffer();
+
+        return true;
+    }
+
+    TEST_CLASS_CLEANUP(ClassCleanup)
+    {
+        m_state->CleanupGlobalScreenBuffer();
+        m_state->CleanupGlobalFont();
+        m_state->CleanupGlobalInputBuffer();
+
+        m_state.release();
+
+        return true;
+    }
+
+    TEST_METHOD_SETUP(MethodSetup)
+    {
+        // STEP 1: Set up the Terminal
+        term = std::make_unique<Terminal>();
+        term->Create({ CommonState::s_csBufferWidth, CommonState::s_csBufferHeight }, 0, emptyRT);
+
+        // STEP 2: Set up the Conpty
+
+        // Set up some sane defaults
+        auto& g = ServiceLocator::LocateGlobals();

```

Comment:
Oh i am very deeply concerned that the TerminalCore tests have a dependency on `CONSOLE_INFORMATION` 😦

Reply:
I mean if it's going to straddle looking at conpty and terminal... it sort of has to. The alternative is breaking it into YET ANOTHER library.

Reply:
Yea it's kinda the _whole point_ of this particular test. Yes I _could_ move this to it's own project, but is that worth it?